pkg/dockerregistry/server/registry.go

```diff
@@ -0,0 +1,23 @@
+package server
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/middleware/registry"
+)
+
+// dockerRegistry represents a collection of repositories, addressable by name.
+// This variable holds the object created by the docker/distribution. We import
+// it into our namespace because there are no other ways to access it. In other
+// cases it is hidden from us.
+var dockerRegistry distribution.Namespace
+
+func init() {
+	middleware.Register("openshift", func(ctx context.Context, registry distribution.Namespace, options map[string]interface{}) (distribution.Namespace, error) {
+		log.Info("Middleware for registry")
```

Comment:
I did it on purpose to see what middlewares have been initialized. This is very important in log analysis because if any of the middlewares will not be initialized, the server will return 500 and understand why this is happening will be very difficult.